apps/dashboard/src/components/editor/use-editor-content.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * External dependencies
+ */
+import { useCallback, useEffect, useState } from '@wordpress/element';
+import { useDispatch } from '@wordpress/data';
+import { debounce } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { STORE_NAME } from '../../data';
+import UnpublishedChangesNotice from './unpublished-changes-notice';
+import { EDITOR_VIEW_DRAFT } from '../../util/editor';
+
+// Do something about the timeout - if no project ID should be 500 ?
+export const useEditorContent = ( projectId, editorView ) => {
+	const [ ready, setReady ] = useState( ! projectId );
+
+	const { removeNotice } = useDispatch( 'core/notices' );
+	const {
+		updateEditorContent,
+		restoreEditorContent,
+		setEditorContentChanged,
+	} = useDispatch( STORE_NAME );
+
+	// If projectId or editorView changes, reset 'ready'
+	useEffect( () => {
+		setReady( false );
+	}, [ projectId, editorView ] );
+
+	return useCallback(
+		// using debounce here so we don't pound the state synchronization
+		debounce( ( content ) => {
+			// Isolated block editor forces a save as soon as the editor content has loaded.
+			// Ignore the first save and set 'ready' to true.
+			//
+			// https://github.com/Automattic/isolated-block-editor/blob/bca504ae1ef98cf1aba136d70e29fc339aa8ec61/src/components/content-saver/index.js#L47
+			if ( ! ready ) {
+				// editorView changes via setForceDraft to EDITOR_VIEW_DRAFT, meaning a restore intent
+				if ( editorView === EDITOR_VIEW_DRAFT ) {
+					restoreEditorContent( content );
```

Comment:
Why do we need this? As in, you can only restore the draft if you haven't made any changes. Meaning, the editor content state would also be empty at that point. What am I missing?

Reply:
Well, when we want to restore the content we use the action from the notice to `setForceDraft( true )`, which changes `editorView` parameter from `auto` to `draft` and picks the draft content over the public one. This param is being *watched* by the `useEffect` at line 26 and, as it changes, sets `ready` to false, triggering the re-render/eval of the hook. When we get to this point with the 2 conditions (ready === false && editorView === draft) it means a restore is going on, so we restore the content, set ready to true and abort the callback.
But now I'm guessing you knew this and maybe your question aims to some different explanation?

Anyways, something broke and I'm not getting the desired effect (restore should "rollback" to draft and leave the app in a "saved" state, meaning the "Save draft" button should be disabled. This was happening when I pushed the changes and now it's not, need to figure it out.

This is also valid for your comment/question above:

> I guess now that we have actions for saving the actual changes, we don't need setEditorContentChanged() anymore - the reducer can just update the flag when any of the other actions is dispatched.

I think it all started to fail now that I've removed the `debounce`d call.